src/Carnac.Logic/Models/Message.cs

```diff
@@ -46,6 +49,13 @@ public Message(IEnumerable<KeyPress> keys, KeyShortcut shortcut)
                 throw new InvalidOperationException("Keys are from different processes");
 
             ProcessName = distinctProcessName.Single();
+
+            var distinctProcessIcon = keys.Select(k => k.Process.ProcessIcon).Distinct();
```

Comment:
Probably do not need this check as the one with the name above should always catch this case?